packages/ui-keyring/src/index.ts

```diff
@@ -7,36 +7,28 @@ import { SingleAddress } from './observable/types';
 import { KeyringAddress, KeyringInstance, KeyringJson, KeyringJson$Meta, State } from './types';
 
 import store from 'store';
-import testKeyring from '@polkadot/keyring/testing';
 import createPair from '@polkadot/keyring/pair';
+import testKeyring from '@polkadot/keyring/testing';
 import { decodeAddress, encodeAddress } from '@polkadot/keyring';
 import { hexToU8a, isHex, isString } from '@polkadot/util';
 
-import observableAccounts from './observable/accounts';
-import observableAddresses from './observable/addresses';
+import accounts from './observable/accounts';
+import addresses from './observable/addresses';
 import observableDevelopment from './observable/development';
 import { accountKey, addressKey, accountRegex, addressRegex, MAX_PASS_LEN } from './defaults';
 import keyringOption from './options';
 
-// FIXME The quicker we get in https://github.com/polkadot-js/apps/issues/138
-// the better, this is now completely out of control
-class Keyring implements KeyringInstance {
-  private state: State;
-
-  constructor () {
-    // state must be always defined in the constructor even if trying to create second instance to overcome TSLint error
-    this.state = {
-      accounts: observableAccounts,
-      addresses: observableAddresses,
-      keyring: testKeyring()
-    };
-
-    // NOTE Everything is loaded in API after chain is received
-    // this.loadAll();
-  }
+const state: State = {
```

Comment:
We could just have these on `this`. (Which is closer to OOP). Then we just go `this.keyring` instead of `const { keyring } = state;`

So we just need some small hoops for `keyring` itself since it is actually not initialised in the constructor and we will make the linter unhappy. I suggest the following -

```
class Keyring ... {
  private _keyring: KeyRingInstance: // not 100% sure of the type

  get keyring (): KeyringInstance {
    if (this._keyring) {
      return this._keyring;
    }

    throw new Error('Keyring should be initialised via 'loadAll' before use');
  } 

  ...
  someFunction () {
    this.keyring.doSomething();
  }
}
```